nautobot/circuits/navigation.py

```diff
@@ -0,0 +1,105 @@
+from nautobot.extras.nautobot_app import NavMenuButton, NavMenuItem, NavMenuTab, NavMenuGroup
+from nautobot.utilities.choices import ButtonColorChoices
+
+
+menu_tabs = (
+    NavMenuTab(
+        name="Circuits",
+        weight=500,
+        groups=(
+            NavMenuGroup(
+                name="Circuits",
+                weight=100,
+                items=(
+                    NavMenuItem(
+                        link="circuits:circuit_list",
+                        link_text="Circuits",
+                        permissions=[
+                            "circuits.view_circuit",
+                        ],
+                        buttons=(
+                            NavMenuButton(
+                                link="circuits:circuit_add",
+                                title="Circuits",
+                                icon_class="mdi mdi-plus-thick",
+                                color=ButtonColorChoices.GREEN,
```

Comment:
It'd be great to have subclasses of `NavMenuButton` that provide defaults for the repeated Add/Import button pattern, something like:

```python
class NavMenuAddButton(NavMenuButton):
    def __init__(*args, **kwargs):
        if "title" not in kwargs:
            kwargs["title"] = "Add"
        if "icon_class" not in kwargs:
            kwargs["icon_class"] "mdi mdi-plus-thick",
        if "color" not in kwargs:
            kwargs["color"] = ButtonColorChoices.GREEN
        super().__init__(*args, **kwargs)

class NavMenuImportButton(NavMenuButton):
    def __init__(*args, **kwargs):
        if "title" not in kwargs:
            kwargs["title"] = "Import"
        if "icon_class" not in kwargs:
            kwargs["icon_class"] "mdi mdi-database-import-outline",
        if "color" not in kwargs:
            kwargs["color"] = ButtonColorChoices.BLUE
        super().__init__(*args, **kwargs)
```

Then we could do the much briefer:

```python
buttons = (
    NavMenuAddButton(link="circuits:circuit_add", permissions=["circuits.add_circuit"]),
    NavMenuImportButton(link="circuits:circuit_import", permissions=["circuits.add_circuit"])
),
```

Reply:
This could potentially be extended further to have a subclass of `NavMenuItem` to which you provide only the model (and perhaps a `weight`) and it automatically constructs a link to the `list` url with the appropriate `view` permission plus the corresponding `Add` and `Import` buttons...